src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/DataSetFileMetaMigrationController.cs

```diff
@@ -0,0 +1,94 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Models;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api;
+
+[Route("api")]
+[ApiController]
+[Authorize(Roles = GlobalRoles.RoleNames.BauUser)]
+public class DataSetFileMetaMigrationController(ContentDbContext contentDbContext) : ControllerBase
+{
+    public class DataSetFileMetaMigrationResult
+    {
+        public bool IsDryRun;
+        public int Processed;
+    }
+
+    [HttpPut("bau/migrate-datasetfilemeta")]
+    public async Task<DataSetFileMetaMigrationResult> MigrationDataSetFileMeta(
+        [FromQuery] bool dryRun = true,
+        [FromQuery] int? num = null,
+        CancellationToken cancellationToken = default)
+    {
+        var queryable = contentDbContext.Files
+            .Where(f => f.DataSetFileMeta == null
+                && f.DataSetFileMetaOld != null);
+
+        if (num != null)
+        {
+            queryable = queryable.Take(num.Value);
```

Comment:
What happens if `num` exceeds the remaining number of records, does it throw or just return whatever remains?